File: codeforces/1472/1472B.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {
    val In = BufferedReader(InputStreamReader(System.`in`))
    val cases = In.readLine()!!.toInt()
    for(i in 1..cases) {
        var n = In.readLine()!!.toInt()
        val weights = In.readLine()!!.split(" ").map {it.toInt()}
        val ones = weights.filter { it == 1 }.size
        val twos = n - ones
        if(twos % 2 == 0 && ones % 2 == 0)
                println("YES")
        else if(twos % 2 == 1 && ones - 2 >= 0 && (ones - 2) % 2 == 0)
                println("YES")
        else
            println("NO")
    }
}

File: codeforces/1472/B/1472B.cpp
#include<iostream>
#include<vector>
#include<map>
#include<algorithm>
using namespace std;

int main(){
    int t;
    scanf(" %d",&t);
    while(t--){
        int n;
        scanf(" %d",&n);
        int ones = 0, twos = 0;
        while(n--){
            int val; scanf(" %d",&val);
            if(val == 2) ++twos;
            else ++ones;
        }
        //printf("%d %d\n",ones,twos);
        if(ones&1){puts("NO");continue;}
        if(twos%2 == 0 || (ones > 1)){puts("YES");}
        else puts("NO");
    }
    return 0;
}